app/src/main/java/org/simple/clinic/drugs/search/DrugSearchListItem.kt

```diff
@@ -0,0 +1,149 @@
+package org.simple.clinic.drugs.search
+
+import androidx.paging.PagingData
+import androidx.paging.TerminalSeparatorType.SOURCE_COMPLETE
+import androidx.paging.insertSeparators
+import androidx.paging.map
+import androidx.recyclerview.widget.DiffUtil
+import com.google.android.material.shape.ShapeAppearanceModel
+import io.reactivex.subjects.Subject
+import org.simple.clinic.R
+import org.simple.clinic.databinding.ListItemDrugSearchBinding
+import org.simple.clinic.databinding.ListItemDrugSearchCornerCapBinding
+import org.simple.clinic.drugs.search.DrugSearchListItem.Event
+import org.simple.clinic.widgets.PagingItemAdapter
+import org.simple.clinic.widgets.dp
+import org.simple.clinic.widgets.recyclerview.BindingViewHolder
+
+sealed class DrugSearchListItem : PagingItemAdapter.Item<Event> {
+
+  companion object {
+
+    fun from(searchResults: PagingData<Drug>, searchQuery: String): PagingData<DrugSearchListItem> {
+      return searchResults
+          .map(::DrugSearchResult)
+          .insertSeparators(SOURCE_COMPLETE) { oldItem, newItem ->
+            if (oldItem != null && newItem != null) {
+              Divider
+            } else if (oldItem == null && newItem == null) {
+              NewCustomDrug(name = searchQuery)
+            } else if (oldItem == null && newItem != null) {
+              TopCornerCapItem
+            } else if (oldItem != null && newItem == null) {
+              BottomCornerCapItem
+            } else {
+              null
+            }
+          }
```

Comment:
This logic seems quite complex. Can we add a test to verify that the items are being generated correctly?